Gregorian calendar being mapped to XMLGregorianCalendar

I have run into a very strange issue where a GregorianCalendar is being cast as an XMLGregorianCalendar.  I can't figure out why, and have included a failing test class.  I can submit it as a pull request, but I don't know yet how it would be fixed.

```java
package ma.glasnost.orika.test;

import ma.glasnost.orika.MapperFacade;
import ma.glasnost.orika.MapperFactory;
import ma.glasnost.orika.impl.DefaultMapperFactory;
import org.junit.Test;

import java.util.GregorianCalendar;
import java.util.HashMap;
import java.util.Map;

public class BrokenTest {

    @Test
    public void testConvertingGregorianCalendar() {
        MapperFactory mapperFactory = new DefaultMapperFactory.Builder()
                .mapNulls(false)
                .build();

        mapperFactory.classMap(ArrayOfItems.class, MapObject.class)
                .field("items{name}", "map{key}")
                .field("items{value}", "map{value}")
                .register();

        MapperFacade mapperFacade = mapperFactory.getMapperFacade();

        Item item = new Item("SomeDate", new GregorianCalendar());
        ArrayOfItems arrayOfItems = new ArrayOfItems(new Item[]{item});

        MapObject<String, Object> map = new MapObject<String, Object>();
        mapperFacade.map(arrayOfItems, map);
    }

    public static class ArrayOfItems{
        private Item[] items;

        public ArrayOfItems(Item[] items) {
            this.items = items;
        }

        public Item[] getItems() {
            return items;
        }

        public void setItems(Item[] items) {
            this.items = items;
        }
    }

    public static class Item {
        private String name;
        private Object value;

        public Item(String name, Object value) {
            this.name = name;
            this.value = value;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }

        public Object getValue() {
            return value;
        }

        public void setValue(Object value) {
            this.value = value;
        }
    }

    public static class MapObject<K, V> {

        private Map<K, V> map;

        public MapObject() {
            this.map = new HashMap<K, V>();
        }

        public MapObject(Map<K, V> map) {
            this.map = map;
        }

        public void setMap(Map<K, V> map) {
            this.map = map;
        }

        public Map<K, V> getMap() {
            return map;
        }
    }

}
```

Here is the error I am getting
```
ma.glasnost.orika.MappingException: While attempting the following mapping:
sourceClass = class ma.glasnost.orika.test.BrokenTest$ArrayOfItems
destinationType = ma.glasnost.orika.test.BrokenTest.MapObject<Object, Object>
resolvedStrategy = MapExistingAndUseCustomMapperStrategy<ArrayOfItems, MapObject<Object, Object>> {customMapper: GeneratedMapper<ArrayOfItems, MapObject<Object, Object>> {usedConverters: [builtin:ToStringConverter<Object, Object>], usedMappers: [], usedMapperFacades: [DefaultBoundMapperFacade<Item, MapEntry<Object, Object>>, DefaultBoundMapperFacade<Object, Object>, DefaultBoundMapperFacade<Entry<Object, Object>, Item>], usedTypes: [Item[], ArrayList<MapEntry<Object, Object>>, String, Map<Object, Object>, ArrayList<Item>] }, unenhancer: ma.glasnost.orika.unenhance.BaseUnenhancer@4e41089d}
Error occurred: java.lang.ClassCastException: java.util.GregorianCalendar cannot be cast to javax.xml.datatype.XMLGregorianCalendar

	at ma.glasnost.orika.impl.ExceptionUtility.newMappingException(ExceptionUtility.java:55)
	at ma.glasnost.orika.impl.MapperFacadeImpl.map(MapperFacadeImpl.java:385)
	at ma.glasnost.orika.impl.MapperFacadeImpl.map(MapperFacadeImpl.java:356)
	at ma.glasnost.orika.impl.MapperFacadeImpl.map(MapperFacadeImpl.java:396)
	at ma.glasnost.orika.test.BrokenTest.testConvertingGregorianCalendar(BrokenTest.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.ClassCastException: java.util.GregorianCalendar cannot be cast to javax.xml.datatype.XMLGregorianCalendar
	at ma.glasnost.orika.converter.builtin.DateAndTimeConverters$CalendarToXmlGregorianCalendarConverter.convertFrom(DateAndTimeConverters.java:304)
	at ma.glasnost.orika.converter.BidirectionalConverter.convert(BidirectionalConverter.java:50)
	at ma.glasnost.orika.impl.mapping.strategy.UseConverterStrategy.map(UseConverterStrategy.java:56)
	at ma.glasnost.orika.impl.DefaultBoundMapperFacade.map(DefaultBoundMapperFacade.java:137)
	at ma.glasnost.orika.MapEntry_Item_ObjectFactory233016336473305233016716740218$3.create(MapEntry_Item_ObjectFactory233016336473305233016716740218$3.java)
	at ma.glasnost.orika.impl.DefaultBoundMapperFacade.newObject(DefaultBoundMapperFacade.java:204)
	at ma.glasnost.orika.generated.Orika_MapObject_ArrayOfItems_Mapper233016545174977$0.mapAtoB(Orika_MapObject_ArrayOfItems_Mapper233016545174977$0.java)
	at ma.glasnost.orika.impl.mapping.strategy.UseCustomMapperStrategy.map(UseCustomMapperStrategy.java:77)
	at ma.glasnost.orika.impl.MapperFacadeImpl.map(MapperFacadeImpl.java:367)
	... 25 more
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,9 @@
 sudo: false
 language: java
 jdk:
-  - oraclejdk7
+  - openjdk7
   - oraclejdk8
+  - oraclejdk9
 branches:
   only:
     - master

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,6 @@ language: java
 jdk:
   - openjdk7
   - oraclejdk8
-  - oraclejdk9
 branches:
   only:
     - master

--- a/core/src/main/java/ma/glasnost/orika/converter/builtin/DateAndTimeConverters.java
+++ b/core/src/main/java/ma/glasnost/orika/converter/builtin/DateAndTimeConverters.java
@@ -560,7 +560,7 @@ public class DateAndTimeConverters {
     }
     
     public static boolean _polyCanConvert(Type<?> a, Type<?> b, Type<?> c, Type<?> d) {
-        return (a.isAssignableFrom(c)) && (d.isAssignableFrom(b));
+        return (a.isAssignableFrom(c)) && (!d.getRawType().equals(Object.class) && d.isAssignableFrom(b));
     }
     
     public static boolean polyCanConvert(Type<?> a, Type<?> b, Type<?> c, Type<?> d) {

--- a/tests/src/main/java/ma/glasnost/orika/test/BrokenTest.java
+++ b/tests/src/main/java/ma/glasnost/orika/test/BrokenTest.java
@@ -1,0 +1,21 @@
+package ma.glasnost.orika.test;
+
+import ma.glasnost.orika.MapperFacade;
+import ma.glasnost.orika.MapperFactory;
+import ma.glasnost.orika.impl.DefaultMapperFactory;
+import org.junit.Test;
+
+import java.util.GregorianCalendar;
+
+public class BrokenTest {
+
+    @Test
+    public void testConvertingGregorianCalendar() {
+        MapperFactory mapperFactory = new DefaultMapperFactory.Builder()
+                .build();
+
+        MapperFacade mapperFacade = mapperFactory.getMapperFacade();
+        GregorianCalendar gc = new GregorianCalendar();
+        Object dest = mapperFacade.map(gc, Object.class);
+    }
+}

